vms/platformvm/state/stakers.go

```diff
@@ -297,6 +297,7 @@ func (s *diffStakers) PutValidator(staker *Staker) {
 		s.addedStakers = btree.NewG(defaultTreeDegree, (*Staker).Less)
 	}
 	s.addedStakers.ReplaceOrInsert(staker)
+	delete(s.deletedStakers, staker.TxID)
```

Comment:
without these changes we won't be able to
Delete unknown staker --> add the staker --> query the staker
cause it will be wrongly marked as deleted.

This flow does not happen in production today as tx verifications forbids it. But it makes the diffs and state more similar and easier to reason about